core/src/nodes/listeners.rs

```diff
@@ -104,15 +104,15 @@ where
         /// The produced upgrade.
         upgrade: TTrans::ListenerUpgrade,
         /// Address of the listener which received the connection.
-        listen_addr: Multiaddr,
+        listen_addrs: MultiaddrSeq,
```

Comment:
Well, that's clearly wrong. A connection only arrives on a single multiaddress. That means more changes are needed.